### DISCUSSION TITLE
Auto Capture Snapshot

### DISCUSSION THREAD

**Discussion by Fotis112** [mcw0/DahuaConsole#21] [Ideas]
Looks like the users in this GitHub really knows workarounds for any situation. 
Is it possible to configure the system to capture a snapshot right after the doorbell button is pushed?

**Comment by riogrande75** [comment]
I think this question does not really fit in here.
Pls. have a look at [https://github.com/riogrande75/Dahua/blob/master/DahuaEventHandler.php](url)
There is a example for taking a snapshot in "function SaveSnapshot".

**Comment by mcw0** [comment]
Primarily (from my side) with this research and code is to research and share some undocumented findings, my upcoming plan (maybe in few weeks from now) is to make changes where it will be very easy to implement own custom functions, like 'modules' alike stuff. It's doable today as well (with some difficulties), but the current code are not so readable as I wish it will be. Will work on that.
